File: android/app/src/main/kotlin/com/swastricare/health/domain/model/MedicationWithSchedule.kt
package com.swastricare.health.domain.model

/**
 * Read-write summary row used by the Family Member Reminders screen
 * (Android Batch J). Joins one `medication_schedules` row with its parent
 * `medications` row so the screen can render the medication name alongside
 * its schedule controls.
 *
 * @property scheduleType One of `daily`, `weekly`, `monthly`, `as_needed`,
 *                        `custom`. Only `daily` is editable in v1.
 * @property timeOfDay `HH:mm:ss` (Postgres TIME). Display as `HH:mm`.
 */
data class MedicationWithSchedule(
    val medicationId: String,
    val medicationName: String,
    val scheduleId: String,
    val scheduleType: String,
    val timeOfDay: String,
    val frequencyPerDay: Int,
    val reminderEnabled: Boolean,
    val isActive: Boolean,
)

File: swastricare-mobile-swift/Models/Family/MedicationWithSchedule.swift
//
//  MedicationWithSchedule.swift
//  swastricare-mobile-swift
//
//  Family Monitoring — Batch 6.
//  Flattened view-model for a medication + one of its active schedules,
//  used by FamilyMemberRemindersView for remote editing by caregivers.
//

import Foundation

struct MedicationWithSchedule: Identifiable, Codable, Equatable {
    var id: String { scheduleId }
    let medicationId: String
    let medicationName: String
    let scheduleId: String
    let scheduleType: String       // 'daily' / 'weekly' / 'monthly' / 'as_needed' / 'custom'
    let timeOfDay: String          // "HH:mm:ss"
    let frequencyPerDay: Int
    let reminderEnabled: Bool
    let isActive: Bool
}
